src/Widgets/UserBox.vala

```diff
@@ -22,54 +22,60 @@ public class Session.Widgets.Userbox : Gtk.ListBoxRow {
     private const string LOGGED_OFF = _("Logged out");
     private const int ICON_SIZE = 48;
 
-    public Act.User? user { public get; private set; }
-    public bool is_guest = false;
+    public Act.User? user { get; set; }
+    public bool is_guest { get; set; default = false; }
+    public string fullname { get; set; }
 
     private Granite.Widgets.Avatar avatar;
     private Gtk.Label fullname_label;
     private Gtk.Label status_label;
 
     public Userbox (Act.User user) {
-        this.user = user;
+        Object (user: user);
         build_ui ();
-        connect_signals ();
-        update ();
-        update_state ();
     }
 
     public Userbox.from_data (string fullname, bool logged_in, bool is_guest = false) {
-        this.is_guest = is_guest;
-        this.user = null;
+        Object (fullname: fullname,
+                is_guest: is_guest,
+                user: null);
         build_ui ();
-        fullname_label.label = "<b>" + fullname + "</b>";
-        update_state ();
     }
 
```

Comment:
You should probably remove the `build_ui` method entirely and put everything in it to `construct`. I also think this is managable in this PR.